### PR TITLE
ENH add broken to removals

### PR DESCRIPTION
**Pull request by beckermr**
cc @isuruf @brianv0

Checklist
* [ ] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge-admin rerender

File: recipe/gen_patch_json.py [hunk 1]
```diff
@@ -289,6 +311,7 @@ def _gen_patch_instructions(index, new_index, subdir):
         "remove": [],
     }
 
+    _add_broken_to_removals()
```
Comment (isuruf) [inline recipe/gen_patch_json.py:314]: This should only be done for `subdir`

Comment (beckermr) [reply recipe/gen_patch_json.py:314]: This call is in the diff function and so it only happens per subdir.

Comment (beckermr) [reply recipe/gen_patch_json.py:314]: See the function implementation itself.

File: recipe/gen_patch_json.py [hunk 2]
```diff
@@ -281,6 +283,26 @@
     'openmpi-4.0.1-hc99cbb1_2': '10.12',
 }
 
+
+def _add_broken_to_removals():
+    global REMOVALS
+    for subdir in SUBDIRS:
```
Comment (beckermr) [inline recipe/gen_patch_json.py:289]: See here @isuruf

File: recipe/gen_patch_json.py (hunk 1 above)
Comment (isuruf) [reply recipe/gen_patch_json.py:314]: That's redundant downloading and computations.

Comment (beckermr) [reply recipe/gen_patch_json.py:314]: Sure I can fix that.

**Comment by beckermr** [comment]
ready for another look @isuruf !

File: recipe/gen_patch_json.py [hunk 3]
```diff
@@ -281,6 +283,25 @@
     'openmpi-4.0.1-hc99cbb1_2': '10.12',
 }
 
+
+def _add_broken_to_removals(subdir):
+    global REMOVALS
+    r = requests.get(
+        "https://conda-static.anaconda.org/conda-forge/"
```
Comment (isuruf) [inline recipe/gen_patch_json.py:290]: Shouldn't this be `conda-web` ?

Comment (beckermr) [reply recipe/gen_patch_json.py:290]: Idk the difference between these things. I asked in the core gitter.

Comment (beckermr) [reply recipe/gen_patch_json.py:290]: We should make sure anaconda is ok with us using conda-web here too

File: recipe/gen_patch_json.py [hunk 4]
```diff
@@ -284,7 +284,7 @@
 }
 
 
-def _add_broken_to_removals(subdir):
+def _add_removals(instructions, subdir):
     global REMOVALS
```
Comment (beckermr) [inline recipe/gen_patch_json.py:288]: We can remove this global call I think

**Comment by beckermr** [comment]
This way of doing things broke the tests because the tests assume only the stuff in the removals global gets added.

**Comment by beckermr** [comment]
@isuruf can you push a fix or back out your change?

File: recipe/test_gen_patch_json.py [hunk 5]
```diff
@@ -26,7 +26,7 @@ def test_gen_patch_instructions():
         'a': {'depends': ['c', 'd', 'e'], 'features': None},
         'c': {'addthis': 'yes'}}
 
-    assert inst['remove'] == list(REMOVALS['osx-64'])
+    assert set(REMOVALS['osx-64']) in set(inst['remove'])
```
Comment (beckermr) [inline recipe/test_gen_patch_json.py:29]:
```suggestion
    assert set(REMOVALS['osx-64']) <= set(inst['remove'])
```

File: recipe/gen_patch_json.py [hunk 6]
```diff
@@ -281,6 +283,24 @@
     'openmpi-4.0.1-hc99cbb1_2': '10.12',
 }
 
+
+def _add_removals(instructions, subdir):
+    r = requests.get(
+        "https://conda-static.anaconda.org/conda-forge/"
```
Comment (beckermr) [inline recipe/gen_patch_json.py:289]:
```suggestion
        "https://conda.anaconda.org/conda-forge/"
```

**Comment by beckermr** [comment]
LGTM!

**Comment by jakirkham** [comment]
That's really cool! Thanks for adding this 😄

**Comment by beckermr** [comment]
Well this seems fine to me given the comments we have gotten. ready to merge @isuruf?

**Comment by beckermr** [comment]
Thanks!

**Pull request merged**